src/v0/destinations/intercom/transform.js

```diff
@@ -93,17 +93,24 @@ function validateTrack(payload) {
   return payload;
 }
 
-function checkIfEmailOrUserIdPresent(message) {
-  return !!(message.userId || message.context?.traits?.email);
-}
+const checkIfEmailOrUserIdPresent = (message, Config) => {
+  let user_id = message.userId;
```

Comment:
For what purpose is this additional check added ? I don't see usage of anonymousId in `attachUserAndCompany` function.
What about the scenario of both userId & email not being available but sendAnonymousId is set & anonymousId being in the payload ?

Reply:
user_id set to anonymousId in attachUserAndCompany function when sendAnonymousId toggle is enabled and userId is not present in payload.
Line no. 112